ccf_transaction_processor/transaction_processor/pdo_tp.cpp

```diff
@@ -51,6 +54,32 @@ namespace ccfapp
             return ccf::make_success(true);
         };
 
+        //======================================================================================================
+        // register enclave expected measurements (member method)
+        auto set_contract_enclave_attestatation_verification_policy = [this](auto& ctx, const nlohmann::json& params) {
+
+            const auto in = params.get<RegisterContractEnclaveAttestationVerificationPolicy::In>();
+
+            // get the current measurements view
+            auto attestation_policy_view = ctx.tx.rw(attestation_policy_table);
+
+            // collect the data to be stored
+            ContractEnclaveAttestationVerificationPolicy attestation_policy;
+
+            attestation_policy.check_attestation = in.check_attestation;
+            if(in.check_attestation){
+                attestation_policy.mrenclave = in.mrenclave;
+                attestation_policy.basename = in.basename;
+                attestation_policy.ias_public_key = in.ias_public_key;
+            }
+
+            //store the data
+            attestation_policy_view->put(PDO_ENCLAVE_ATTESTATION_POLICY, attestation_policy);
+
+            return ccf::make_success(true);
```

Comment:
what happens when we get multiple requests to set the policy? it looks like you just overwrite it.  but what about enclaves verified with the old policy? are they still valid? how does someone using the ledger know which policy was used to accept the enclaves? the simplest policy would be to just disallow changes once it is set.

Reply:
yes, new policy replaces old policy. IMHO, that is totally reasonable to assume that CCF consortium gets to change the policy when they see fit. (eg.g, when Intel updates revocation list, MREnclave is regenerated, and updated). Yes, agreed that there should be two additional changes to TP 1. A new rpc that can be used by endpoints to determine current/past policies 2. verify_enclave_registration rpc must return the policy id used to verify the enclave. This is a feature improvement, and can be taken up as a separate PR, while retaining the current code in this PR.